File: Algorithms/Warmup/Lonely Integer/LonelyInteger.java
import java.util.Scanner;
import java.util.ArrayList;

public class LonelyInteger{
	public static int lonely_integer(ArrayList<Integer> arr, int array_size){
		int res = 0;
		int i;
		for(i = 0; i < array_size; i++){
			res = res ^ arr.get(i);
		}
		return res;
	}


	public static void main(String[] args){
		
		ArrayList<Integer> arr = new ArrayList<Integer>();
		int array_size;
		int i, result;
		
		Scanner in = new Scanner(System.in);
		array_size = in.nextInt();

		for(i = 0; i < array_size; i++){
			arr.add(in.nextInt());
		}

		result = lonely_integer(arr, array_size);
		System.out.print(result);


	}
}

File: Algorithms/Warmup/Lonely Integer/lonely_integer.cpp
#include <cstdlib>
#include <iostream>

int lonely_integer(int* arr, int array_size){
	int res = 0;
	size_t i;

	for(i = 0; i < array_size; i++){
		res = res ^ arr[i];
	}

	return res;

}

int main(){

	int array_size;
	int* arr;
	int result;
	int i;
	std::cin >> array_size;
	arr = (int*)malloc(array_size*sizeof(int));

	for(i = 0; i < array_size; i++){
		std::cin >> arr[i];
	}

	result = lonely_integer(arr, array_size);
	std::cout << result;
	return 0;
}